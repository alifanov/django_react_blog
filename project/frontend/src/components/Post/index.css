.post-tags {
  font-size: 13px;
  margin-bottom: 10px;
}
