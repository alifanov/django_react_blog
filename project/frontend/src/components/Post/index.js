import React from "react";
import PropTypes from "prop-types";

import { Card, Button, CardBody, CardText, CardHeader } from "reactstrap";
import { FaTimesCircle } from "react-icons/fa";
import Moment from "react-moment";

import "./index.css";

const Post = props => {
  return (
    <Card className="mb-4 post-item">
      <CardHeader className={"d-flex justify-content-between"}>
        <div className={"d-flex align-items-center"}>{props.title}</div>
        <FaTimesCircle
          onClick={() => props.onDelete(props.id)}
          style={{ cursor: "pointer" }}
        />
      </CardHeader>
      <CardBody>
        <CardText>{props.text}</CardText>
        <div className="post-tags">
          {props.tags.map(t => (
            <span key={t}>#{t}</span>
          ))}
        </div>
        <div className="d-flex justify-content-between align-items-center">
          <div className="btn-group">
            <Button size={"sm"} outline color={"secondary"}>
              Read
            </Button>
          </div>
          <small className="text-muted">
            Published:&nbsp;
            <Moment format={"DD.MM.YYYY HH:mm"}>{props.created_at}</Moment>
          </small>
        </div>
      </CardBody>
    </Card>
  );
};

Post.propTypes = {
  onDelete: PropTypes.func
};

export default Post;
