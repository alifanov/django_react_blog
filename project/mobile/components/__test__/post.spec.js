import React from "react";
import { View, Text } from "react-native";
import { shallow } from "../../enzyme";
import Post from "../post";

import renderer from "react-test-renderer";

describe(">>> Post test", () => {
  it("+++ render Post component", () => {
    const post = shallow(<Post title={"Title"} text={"Text"} />);
    expect(post.length).toEqual(1);
  });
  it("+++ check text for Post", () => {
    const post = shallow(<Post title={"Title"} text={"Text"} />);
    expect(
      post
        .find(Text)
        .at(0)
        .props().children
    ).toEqual("Title");
    expect(
      post
        .find(Text)
        .at(1)
        .props().children
    ).toEqual("Text");
  });
});

describe(">>>Post --- Snapshot", () => {
  it("+++capturing Snapshot of Post", () => {
    const renderedValue = renderer
      .create(<Post title={"Title"} text={"Text"} />)
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
