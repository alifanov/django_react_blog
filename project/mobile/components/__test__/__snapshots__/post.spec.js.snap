// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Post --- Snapshot +++capturing Snapshot of Post 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "gray",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "marginVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#333",
        "borderBottomWidth": 1,
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ddd",
        }
      }
    >
      Title
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ddd",
        }
      }
    >
      Text
    </Text>
  </View>
</View>
`;
